app/src/main/java/com/soen490chrysalis/papilio/viewModel/UserProfileViewModel.kt

```diff
@@ -0,0 +1,83 @@
+package com.soen490chrysalis.papilio.viewModel
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.google.firebase.auth.EmailAuthProvider
+import com.google.firebase.auth.FirebaseAuth
+import com.soen490chrysalis.papilio.repository.users.IUserRepository
+import kotlinx.coroutines.launch
+
+class UserProfileViewModel(private val userRepository : IUserRepository) : ViewModel()
+{
+    var userObject : MutableLiveData<GetUserResponse> = MutableLiveData<GetUserResponse>()
+    var passwordChangeResult : MutableLiveData<String> = MutableLiveData<String>()
+    private var editedFields : MutableMap<String, kotlin.Any> = mutableMapOf<String, kotlin.Any>()
+
+    fun getUserByFirebaseId()
+    {
+        viewModelScope.launch {
+            val userResponse = userRepository.getUserByFirebaseId()
+            userObject.value = GetUserResponse(true, userResponse?.user)
+        }
+    }
+
+    fun addEditedField(fieldName: String, fieldValue: kotlin.Any)
+    {
+        editedFields.put(fieldName, fieldValue);
```

Comment:
We code in kotlin not in java so you should use kotlin semantics for maps like this:
![image](https://user-images.githubusercontent.com/64218770/211125793-0e9164d0-f194-485c-9a60-110f4fa88f95.png)